### ISSUE TITLE
MoPub stopped showing ads when clicking the ads banner on iOS

### ISSUE THREAD

**Issue by csbubbles** [mopub/mopub-ios-sdk#130]
I cannot open any ad when clicking the MoPub banner. When I click first time, it shows the loading indicator (same way as if it was on iPhone Simulator), then it disappears and nothing shows up. When you click after that, nothing happens.

PS It worked before (last time I have checked it a month or two ago). It was opening Apple Store for games and stuff like that just fine.

iPhone 4S, iPhone 5S
iOS 8.3

iPad
iOS 8.3

**Comment by csbubbles** [comment]
PS On Android everything still works with no issues.

**Comment by csbubbles** [comment]
The only logs I am getting from MoPub:

```
2015-05-07 13:32:46.893 MOPUB: Banner view (...) loading ad with MoPub server URL: http://ads.mopub.com/m/ad?v=8 ... &dn=iPhone4%2C1
2015-05-07 13:32:47.437 MOPUB: Banner ad view is fetching ad network type: html
2015-05-07 13:32:47.443 MOPUB: Looking for custom event class named MPHTMLBannerCustomEvent.
2015-05-07 13:32:47.444 MOPUB: Loading MoPub HTML banner
2015-05-07 13:32:48.444 MOPUB: MoPub HTML banner did load
2015-05-07 13:32:55.274 MOPUB: MoPub HTML banner will begin action
```

**Comment by edwardho** [comment]
Hello csbubbles,

Could you email support@mopub.com with your issue and your MoPub account username and the affected ad unit ID?

This will allow us to troubleshoot your specific issue.

Thanks,
Edward

**Comment by alicerp** [comment]
Thanks, Edward.  Please note that the MoPub Github Issue Reporting Section will now be housed directly on the Twitter Developer Community Forum.

Please visit https://twittercommunity.com/c/fabric/mopub for any additional support questions or feel free to email suport@mopub.com.

**Comment by csbubbles** [comment]
Let's close the issue. I have contacted to the support team, it wasn't really helpful, but I have figured out the issue myself in the end. MoPub doesn't open the ads target if there is no controller callback implemented. Though it still shows the ads and a popup window when you click those ads. I would request to update the developer docs to emphasize how important that callback is and what it may affect. Also, that would be nice to see a warning in logs if there is no callback implemented. Otherwise, you have no idea what is happening.
